packages/types/src/packages/api-reference-configuration.ts

```diff
@@ -0,0 +1,265 @@
+import { z } from 'zod'
+
+import type { TargetId } from '../external/index.ts'
+
+/** Available theme presets for the API reference */
+const ThemeIdEnum = z.enum([
+  'alternate',
+  'default',
+  'moon',
+  'purple',
+  'solarized',
+  'bluePlanet',
+  'deepSpace',
+  'saturn',
+  'kepler',
+  'elysiajs',
+  'fastify',
+  'mars',
+  'none',
+])
+
+/** Valid keys that can be used with CTRL/CMD to open the search modal */
+const SearchHotKeyEnum = z.enum([
+  'a',
+  'b',
+  'c',
+  'd',
+  'e',
+  'f',
+  'g',
+  'h',
+  'i',
+  'j',
+  'k',
+  'l',
+  'm',
+  'n',
+  'o',
+  'p',
+  'q',
+  'r',
+  's',
+  't',
+  'u',
+  'v',
+  'w',
+  'x',
+  'y',
+  'z',
+])
+
+/** Supported integration types */
+const IntegrationEnum = z
+  .enum([
+    'adonisjs',
+    'docusaurus',
+    'dotnet',
+    'elysiajs',
+    'express',
+    'fastapi',
+    'fastify',
+    'go',
+    'hono',
+    'html',
+    'laravel',
+    'litestar',
+    'nestjs',
+    'nextjs',
+    'nitro',
+    'nuxt',
+    'platformatic',
+    'react',
+    'rust',
+    'vue',
+  ])
+  .nullable()
+
+/** Configuration for the OpenAPI/Swagger specification */
+const SpecConfigurationSchema = z.object({
+  /** URL to an OpenAPI/Swagger document */
+  url: z.string().url().optional(),
+  /**
+   * Directly embed the OpenAPI document.
+   * Can be a string, object, function returning an object, or null.
+   * @remarks It's recommended to pass a URL instead of content.
+   */
+  content: z.union([z.string(), z.record(z.any()), z.function().returns(z.record(z.any())), z.null()]).optional(),
+})
+
+/** Configuration for path-based routing */
+const PathRoutingSchema = z.object({
+  /** Base path for the API reference */
+  basePath: z.string(),
+})
+
+export const ApiReferenceConfigurationSchema = z.object({
+  /** A string to use one of the color presets */
+  theme: z.union([ThemeIdEnum, z.undefined()]).optional(),
+  /** The layout to use for the references */
+  layout: z.enum(['modern', 'classic']).optional(),
```

Comment:
```suggestion
  layout: z.enum(['modern', 'classic']).optional().default('modern').catch('modern'),
```
lets set a default wherever we can